### PR TITLE
[CWS] Generate fields resolver function

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR generates the `fields_resolver.go` file from the `model.Event`.

### Motivation

Make sure we don't forget to resolve a field when we need to copy parts of a `model.Event` instance.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/secl/compiler/generators/accessors/common/types.go [hunk 1]
```diff
@@ -5,6 +5,19 @@
 
 package common
 
+// EventTypeMetada is used to iterate over the model from the event types
+type EventTypeMetada struct {
```
Comment (lebauce) [inline pkg/security/secl/compiler/generators/accessors/common/types.go:9]: s/Metada/Metadata ?

Comment (Gui774ume) [reply pkg/security/secl/compiler/generators/accessors/common/types.go:9]: 👍🏻

File: pkg/security/probe/fields_resolver.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	_ = ev.ResolveFileFieldsGroup(&ev.ProcessContext.Process.FileFields)
+	_ = ev.ResolveFileFieldsInUpperLayer(&ev.ProcessContext.Process.FileFields)
+	_ = ev.ResolveFileFieldsUser(&ev.ProcessContext.Process.FileFields)
+
+	// resolve event specific fields
+	switch ev.GetEventType().String() {
+
+	case "bpf":
+		_ = ev.ResolveHelpers(&ev.BPF.Program)
+		break
```
Comment (safchain) [inline pkg/security/probe/fields_resolver.go:30]: that is a `C` thing :), you can remove the `break`

File: pkg/security/probe/fields_resolver.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
+//go:build linux
+// +build linux
+
+// Code generated - DO NOT EDIT.
+
+package probe
+
+// ResolveFields resolves all the fields associate to the event type. Context fields are automatically resolved.
+func (ev *Event) ResolveFields() error {
```
Comment (safchain) [inline pkg/security/probe/fields_resolver.go:9]: we could remove the error return ?

File: pkg/security/probe/fields_resolver.go (hunk 2 above)
Comment (Gui774ume) [reply pkg/security/probe/fields_resolver.go:30]: 👍🏻

File: pkg/security/probe/fields_resolver.go (hunk 3 above)
Comment (Gui774ume) [reply pkg/security/probe/fields_resolver.go:9]: 👍🏻

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**